rholang/src/main/scala/coop/rchain/rholang/interpreter/util/AddressTools.scala

```diff
@@ -0,0 +1,61 @@
+package coop.rchain.rholang.interpreter.util
+
+import java.nio.charset.StandardCharsets
+
+import coop.rchain.crypto.hash.Blake2b256
+import coop.rchain.rholang.interpreter.util.codec.Base58
+
+final case class Address(val prefix : Array[Byte], val heyHash : Array[Byte])
+
+class AddressTools(prefix : Array[Byte], keyLength : Int, checksumLength : Int) {
+  private def computeChecksum(toCheck : Array[Byte]): Array[Byte] =
+    Blake2b256.hash(toCheck).take(checksumLength)
+
+  def fromPublicKey(pk : Array[Byte]) : String = {
```

Comment:
this (and upstream usage in RevAddress) should return via Option / Either whenever `pk.length != keyLength`. We should not be `throw`-ing assertion errors.

Reply:
We have runtime errors modelled by exceptions or Option or Either and we have algorithm errors where you don't want to deal with the error but instead you assert and move forward.

This is a case of the latter because I expect the code calling the fromPublicKey to make sure that the key has the proper length.

Reply:
> I expect the code calling the fromPublicKey to make sure

That's the primary issue with non-fp/impure code: assuming the context in which it is going to be called (instead of defining the possible contexts using effects and the type system).

I disagree with you. The type, and just the type, should be enough to say if I can call a function or not. In no circumstance should it `throw` and actual exception. This is impure, and without a good reason.

Reply:
fair enough. I've changed the code to return Option[String] instead